explorer/src/components/account/TokenHistoryCard.tsx

```diff
@@ -301,13 +310,41 @@ const TokenTransactionRow = React.memo(
       );
 
     const tokenInstructionNames = instructions
-      .map((ix): string | undefined => {
+      .map((ix, index): string | undefined => {
+        let transactionInstruction;
+        if (details?.data?.transaction?.transaction) {
+          transactionInstruction = intoTransactionInstruction(
+            details.data.transaction.transaction,
+            index
+          );
+        }
+
         if ("parsed" in ix) {
           if (ix.program === "spl-token") {
             return instructionTypeName(ix, tx);
           } else {
             return undefined;
           }
+        } else if (
```

Comment:
I was was originally thinking that we would flatten inner instructions here rather than show higher level instructions like swap / serum. I think this is pretty useful though.